src/App.tsx

```diff
@@ -8,7 +8,52 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { FILTER_STATUS } from './utils/FILTER_STATUS';
+import { getTodos } from './api';
+
+import { Todo } from './types/Todo';
+
+function handleFiltration(data: Todo[], query: string, filterBy: string) {
```

Comment:
The function `handleFiltration` could be optimized. Currently, you are creating a new array with `[...data]` on line 17, then potentially creating another new array with the `.filter` method on lines 22 and 25. This is unnecessary and could be optimized. Instead of creating a new array at the beginning, you can directly use the `.filter` method on `data`. If `filterBy` is not truthy, you can return the original `data` array.